### ISSUE TITLE
rtl-sdr: reception works properly with gain to maximum

### ISSUE THREAD

**Issue by funder7** [marcogrecopriolo/guglielmo#11]
Hi, 

I've just completed the build on ubuntu (21.04). 
The app starts without any problem, but it doesn't find stations with the default setup: I had to set the device gain to `99` (max), in order to find channels. 

Do you thin that this is normal? ([device description here](https://www.409shop.com/409shop_product.php?id=124266))

It would be nice to have some kind of quick frequency navigation, to increase by 50 Mhz, or set the frequency tuning step value (it's a bit slow).

DAB Reception works well, but with max gain, I see around 15 stations (welle.io finds around ~30).

**Comment by marcogrecopriolo** [comment]
Hi, my experience with https://www.amazon.co.uk/gp/product/B01GDN1T4S/ref=ppx_yo_dt_b_asin_image_o02_s00?ie=UTF8&psc=1 is that the coil aerial is poor an I have to use the extensible one for decent reception.
I have an external aerial anyway, and I use the default gain of 50.
This said, I have not changed the gains from Jan's original code, but I can certainly experiment with them, and I can also throw a DAB scan feature in.

Ta much,
Marco

**Comment by funder7** [comment]
Hi Marco,
Thanks for the information, I thought that the antenna provided with the device was cheap, but not so ineffective! 
I tried to search for a better solution, but what I've found on Amazon doesn't look much different from the stock one. 

To be honest I'm not an expert about radio communications, so I'm not able to spot which one is better than the other; can you give me some advice on which one to use? 

I have to understand if the 100Khz-30Mhz connector is active or not: tuning on that range of frequencies and connecting/disconnecting the antenna changes absolutely nothing on the spectrum. 
It's not clear how to use the two connectors on the device, they don't send any kind of manual. Actually I don't know if the R820T is working or not, maybe it should be activated by sending a specific command to the device? Who knows.. 

Cheers,
Federico

**Comment by marcogrecopriolo** [comment]
I have reviewed the rtl-sdr library code, and for the R820T tuner (which your device uses), the gain is capped, meaning that if you try to set a gain which is higher than the advertised device gains, rtlsdr_set_tuner_gain() just sets it to the allowed maximum, so there's not much that I can do about that.

The only thing that I can say, is that AGC seems to work well for me, but I use the rtl-sdr hardware AGC.
Welle-io seems to use a software implementation of AGC (which is turned on by default), which may or may not make a difference.
Have you tried to use AGC with guglielmo?

Re aerials, not too much of an expert, sorry.

**Comment by marcogrecopriolo** [comment]
have implemented SW agc - you want to try that?

**Comment by funder7** [comment]
Ok, how?

**Comment by marcogrecopriolo** [comment]
pull the latest source, build it, in the device settings the AGC has now three settings: off, software, on: choose software.

**Comment by funder7** [comment]
Hi, after building the app, I downloaded the latest release to make a comparison, and found software AGC there as well. I guess you already released the feature 😃 

I wasn't able to reproduce the issue, this is a fresh linux install and maybe something changed.
The best audio/reception is with AGC=ON and Gain=50, software AGC works, but the other one seems to return a more "stable" output, even if it's more saturated. 

BTW I need to make a proper comparison by using the same setup I used the last time, when I opened the issue. 
This receiver is weird, I noticed that using a regular USB cable doesn't work very well: using a Y cable improves the reception by increasing the max current to 1A (first usb port is for power+data, the second is for power only).

Cheers
